PR: https://github.com/odoo/odoo/pull/227776

From: eef4fbdfa586171fa0d9ac63160340b0e44a06bd
From: Patricia Daloze (pda)
Date: 2025-09-22 07:00:38

Breaking data model changes scores: del:3 + add:0, change matches:
-    paymob_public_key = fields.Char(string="Paymob Public Key", required_if_provider='paymob')
-    paymob_hmac_key = fields.Char(string="Paymob HMAC Key", required_if_provider='paymob')
-    paymob_api_key = fields.Char(string="Paymob API Key", required_if_provider='paymob')

Total Changes: 24

[IMP] payment_*: set copy=False on credential fields

Since different companies typically require distinct credentials, we
avoid copying them during company creation by setting copy=False on
all credential fields.

Additionally, because credentials are necessary for proper setup,
payment providers should not be published automatically when a new
company is created.

opw-5083526

X-original-commit: 989f058e3ab759d64491b853f4ce11e7f90f4cd8
Part-of: odoo/odoo#227776
Signed-off-by: Antoine Vandevenne (anv) <anv@odoo.com>
Signed-off-by: Patricia Daloze (pda) <pda@odoo.com>

================================= pseudo patch: =================================

--- a/addons/payment_paymob/models/payment_provider.py
+++ b/addons/payment_paymob/models/payment_provider.py
@@ -29,13 +29,29 @@ class PaymentProvider(models.Model):
         inverse='_inverse_paymob_account_country_id',
         domain=f'[("code", "in", {list(const.API_MAPPING.keys())})]',
         required_if_provider='paymob',
+        copy=False,
+    )
+    paymob_public_key = fields.Char(
+        string="Paymob Public Key",
+        required_if_provider='paymob',
+        copy=False,
     )
-    paymob_public_key = fields.Char(string="Paymob Public Key", required_if_provider='paymob')
     paymob_secret_key = fields.Char(
-        string="Paymob Secret Key", required_if_provider='paymob', groups='base.group_system'
+        string="Paymob Secret Key",
+        required_if_provider='paymob',
+        copy=False,
+        groups='base.group_system',
+    )
+    paymob_hmac_key = fields.Char(
+        string="Paymob HMAC Key",
+        required_if_provider='paymob',
+        copy=False,
+    )
+    paymob_api_key = fields.Char(
+        string="Paymob API Key",
+        required_if_provider='paymob',
+        copy=False,
     )
-    paymob_hmac_key = fields.Char(string="Paymob HMAC Key", required_if_provider='paymob')
-    paymob_api_key = fields.Char(string="Paymob API Key", required_if_provider='paymob')
 
     # === CONSTRAINT METHODS === #
 
